openstack/assets/dashboards/openstack-controller.json

```diff
@@ -0,0 +1,693 @@
+            },
+            "time": {
+                "live_span": "1h"
+            },
+            "title": true,
+            "title_align": "left",
+            "title_size": 16,
+            "title_text": "Change in running VMs by hypervisor in $aggregate",
+            "type": "change",
+            "width": 46,
+            "x": 69,
+            "y": 29
+        },
+        {
+            "height": 14,
+            "id": 15,
+            "legend": false,
+            "legend_size": "0",
+            "tile_def": {
+                "requests": [
+                    {
+                        "q": "avg:openstack.nova.current_workload{*} by {hypervisor}",
+                        "type": "bars"
+                    }
+                ],
+                "viz": "timeseries"
+            },
+            "time": {
+                "live_span": "1h"
+            },
+            "title": true,
+            "title_align": "left",
+            "title_size": 16,
+            "title_text": "Nova current workload by hypervisor",
+            "type": "timeseries",
+            "width": 46,
+            "x": 69,
+            "y": 12
+        },
+        {
+            "height": 14,
+            "id": 16,
+            "legend": false,
+            "legend_size": "0",
+            "tile_def": {
+                "requests": [
+                    {
+                        "q": "sum:openstack.nova.vcpus{$aggregate} by {host}",
+                        "style": {
+                            "palette": "warm"
+                        },
+                        "type": "line"
+                    },
+                    {
+                        "q": "sum:openstack.nova.vcpus_used{$aggregate} by {hypervisor}",
+                        "type": "area"
+                    }
+                ],
+                "viz": "timeseries"
+            },
+            "time": {
+                "live_span": "1h"
+            },
+            "title": true,
+            "title_align": "left",
+            "title_size": 16,
+            "title_text": "VCPUs used vs available by hypervisor",
+            "type": "timeseries",
+            "width": 46,
+            "x": 69,
+            "y": 48
+        },
+        {
+            "bgcolor": "gray",
+            "font_size": "36",
+            "height": 7,
+            "html": "[Nova Server](https://www.datadoghq.com/blog/openstack-monitoring-nova/#nova-server-metrics)",
+            "id": 17,
+            "text_align": "center",
+            "tick": true,
+            "tick_edge": "bottom",
+            "tick_pos": "50%",
+            "type": "note",
+            "width": 46,
+            "x": 117,
+            "y": 66
+        },
+        {
+            "height": 14,
+            "id": 18,
+            "legend": false,
+            "legend_size": "0",
+            "tile_def": {
+                "requests": [
+                    {
+                        "q": "avg:openstack.nova.server.hdd_read_req{$project} by {host}",
```

Comment:
`by {host}` won't the cardinality be reaaally high here for real life openstack environments?

Reply:
@FlorianVeaux this dashboard widget is identical to the one in production for OpenStack today. 

The OpenStack controller integration is really only configured on the hypervisors which are relatively few in number.